blackbox_repository/tabulated_benchmark.py

```diff
@@ -12,43 +12,67 @@
 # permissions and limitations under the License.
 from pathlib import Path
 from typing import List, Optional
+import numpy as np
+import logging
 
 from blackbox_repository import load, Blackbox, add_surrogate
+from blackbox_repository.utils import metrics_for_configuration
 from syne_tune.backend.simulator_backend.simulator_backend import SimulatorBackend
 from syne_tune.backend.trial_status import Status
 
+logger = logging.getLogger(__name__)
+
 
 class _BlackboxSimulatorBackend(SimulatorBackend):
 
-    def __init__(self, elapsed_time_attr: str):
+    def __init__(self, elapsed_time_attr: str,
+                 time_this_resource_attr: Optional[str] = None,
+                 max_resource_attr: Optional[str] = None,
```

Comment:
I would prefer to have `time_attr` and `is_cumulative` arguments instead of two strings since the state of two strings may be inconsistent (you may have two None or two strings and one is ignored).  I also believe that the explanation is simpler: "`time_attr` must represent the time spent since the beggining if `is_cumulative` is True or the time spent in the step otherwise."

Reply:
But then, if is_cumulative = False, you need to specify what elapsed_time_attr should be. The problem is that this name is also needed elsewhere, in the schedulers. So, for example, if I somehow construct elapsed_time_attr here, then I'd also have to tell that to schedulers.

Do you have a suggestion?

One thing to note is that it is already a bit of a stretch that a back-end determines what these attribute names are. The other back-ends do not. I see the point here, because due to the BB table, you know the column names. But it is important that also the scheduler needs to know these names, and it cannot just ask the back-end, because normally the back-end would not know.

Reply:
Note that elapsed_time_attr must never be None, this would fail. The simulator back-end does not work if results do not contain this attribute.

Reply:
> The problem is that this name is also needed elsewhere, in the schedulers.

Not 100% clear what you are referring to here, for me schedulers are using `resource_attr` which is typically the epoch but not the runtime as in this case.

I understood your other point: simulator backend currenly only handles cumulative time. I propose to use this code for now, but to add support for adding cumulative time in blackbox in case only a per-step information is present so that we can avoid handling this case here. Does this solution work for you?

Reply:
I mean the cost-aware schedulers. They have an attribute called cost_attr, and for the use cases we have right now, this is equal to elapsed_time_attr.

And elapsed_time_attr is also used with simulation-based scoring (which is not yet merged in). There, you essentially need to train a cost model (like in cost-aware) to fit the training times.

Reply:
Ok, it is a bit independent in my view (the different names seems to indicate that). 

In any case, is the idea of using the code for now, but to add support for adding cumulative time in blackbox later an option? (I could add support for this)